source/find_getmore_killcursors_commands.rst

```diff
@@ -340,9 +340,13 @@ Any driver that provides helpers for any of the special collections below SHOULD
 Exhaust
 ^^^^^^^
 
-The **find** command does not support the exhaust flag from **OP_QUERY**.
-Drivers that support exhaust MUST fallback to existing **OP_QUERY** wire
-protocol messages.
+This section only applies to drivers that support exhaust cursors.
+
+On server versions 3.6 and later drivers MUST use **OP_MSG** for exhaust cursors.
+
+On server versions prior to 3.6, drivers MUST fallback to the legacy **OP_QUERY**
+wire protocol messages. The **find** command does not support the exhaust flag
+from **OP_QUERY**.
```

Comment:
To clarify, is this a server limitation? I checked our code and we have exhaust flag defined for op_query:

```
      # @option options [ Array<Symbol> ] :flags The flag bits.
      #   Currently supported values are :await_data, :exhaust,
      #   :no_cursor_timeout, :oplog_replay, :partial, :secondary_ok,
      #   :tailable_cursor.
```

Reply:
Yes, not supporting the exhaust flag for the find command run over OP_QUERY is a server side limitation.

The `OP_QUERY` wire protocol message has a flag for exhaust, documented here: https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/#op_query. That only applies to the legacy OP_QUERY find.

[There are three ways to send a find](https://docs.google.com/document/d/1dWMHgxO43KwX-HYJQL_F2ivpiko2pj6aZhFS8DtWldo/edit) describes the distinction between the legacy OP_QUERY find and the find command sent over OP_QUERY.